src/Tizen.NUI.WindowSystem/src/public/InputGesture.cs

```diff
@@ -0,0 +1,488 @@
+        }
+
+        /// <summary>
+        /// Destructor.
+        /// </summary>
+        ~InputGesture()
+        {
+            Dispose(false);
+            Log.Debug(LogTag, "InputGesture Destroyed");
+        }
+
+        /// <summary>
+        /// Dispose.
+        /// </summary>
+        public void Dispose()
+        {
+            Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        /// <inheritdoc/>
+        protected virtual void Dispose(bool disposing)
+        {
+            if (!disposed)
+            {
+                Interop.InputGesture.ErrorCode res = Interop.InputGesture.Deinit(gestureHandle);
+                ErrorCodeThrow(res);
+
+                if (disposing) {
+                    gestureHandle?.Dispose();
+                    gestureHandle = null;
+
+                    edgeSwipeData?.Dispose();
+                    edgeSwipeData = null;
+
+                    edgeDragData?.Dispose();
+                    edgeDragData = null;
+
+                    tapData?.Dispose();
+                    tapData = null;
+
+                    palmCoverData?.Dispose();
+                    palmCoverData = null;
+                }
+                disposed = true;
+            }
+        }
+
+        /// <summary>
+        /// Generates a edge swipe gesture's grab info handle
+        /// </summary>
+        /// <param name="fingers"> The number of fingers </param>
+        /// <param name="edge"> The position of edge</param>
+        /// <returns> The edge swipe gesture data handle </returns>
+        public SafeGestureDataHandle EdgeSwipeNew(int fingers, GestureEdge edge)
+        {
+            edgeSwipeData = Interop.InputGesture.EdgeSwipeNew(gestureHandle, fingers, (int)edge);
+            Log.Debug(LogTag, "EdgeSwipeNew");
+            return edgeSwipeData;
+        }
+
+        /// <summary>
+        /// Frees a memory of edge swipe gesture's grab info handle
+        /// </summary>
+        /// <param name="data"> The edge swipe gesture data handle </param>
+        public void EdgeSwipeFree(SafeGestureDataHandle data)
+        {
+            Interop.InputGesture.ErrorCode res = Interop.InputGesture.EdgeSwipeFree(gestureHandle, data);
+            ErrorCodeThrow(res);
+        }
+
+        /// <summary>
+        /// Sets a specific size of edge for edge swipe gesture
+        /// </summary>
+        /// <param name="data"> The edge swipe gesture data handle </param>
+        /// <param name="edgeSize"> The enum of gesture edge size </param>
+        /// <param name="startPoint"> The start point of edge area </param>
+        /// <param name="endPoint"> The end point of edge area</param>
+        public void EdgeSwipeSizeSet(SafeGestureDataHandle data, GestureEdgeSize edgeSize, int startPoint, int endPoint)
+        {
+            Interop.InputGesture.ErrorCode res = Interop.InputGesture.EdgeSwipeSizeSet(data, (int)edgeSize, startPoint, endPoint);
+            ErrorCodeThrow(res);
+        }
```

Comment:
The API you have written is too low-level. Since C# is an object-oriented language, please design the API in an object-oriented manner. If the above API were written in an object-oriented way, it could be redesigned as follows:

``` c#
class EdgeSwipe
{
    public void SetSwipeSize(GestureEdgeSize edgeSize, int startPoint, int endPoint);
}
```